lib/llm-events/error-message.js

```diff
@@ -23,9 +23,24 @@ module.exports = class LlmErrorMessage {
    * @param {LlmTool} [params.tool] Details about the tool event if it was a tool event.
    */
   constructor({ response, cause, summary, embedding, vectorsearch, tool } = {}) {
-    this['http.statusCode'] = response?.status ?? cause?.status
+    // For @google/genai only, cause does not have the `error` or `status` fields,
+    // but it does have `message` with the info we need. So, we need to parse
+    // the relevant fields from cause.message to get `status` and `error`.
+    let parsedError
+    const isGeminiVendor = embedding?.vendor === 'gemini' || summary?.vendor === 'gemini'
+    if (isGeminiVendor && cause?.message) {
+      try {
+        // Extract the JSON portion of the cause.message
+        const jsonStartIndex = cause.message.indexOf('{')
+        const jsonString = cause.message.substring(jsonStartIndex)
+        parsedError = JSON.parse(jsonString)?.error
+      } catch {
+        parsedError = undefined
```

Comment:
Would it be better to pass the `cause.message` as a string, since we know it's defined at this point?
```suggestion
        parsedError = cause.message
```